nsdb-core/src/main/scala/io/radicalbit/nsdb/actors/ShardReaderActor.scala

```diff
@@ -93,17 +93,17 @@ class ShardReaderActor(val basePath: String, val db: String, val namespace: Stri
         schema.tags.headOption.getOrElse(Schema.timestampField -> schema.timestamp))
       quantities <- distinctPrimaryAggregations
         .foldLeft(Try(Map.empty[String, NSDbNumericType])) {
-          case (tryAcc, CountAggregation) =>
+          case (tryAcc, CountAggregation(_)) =>
```

Comment:
In general, I am afraid we may pay `unapply` overhead over time. 
Honestly, I don't know how much `unapply` costs rather than runtime type checking does. But I believe we may figure it out one day so to understand if we can keep going with `unapply`, either moving to reinforced use of runtime type checking (e.g. `case (tryAcc, _: CountAggregation)`)

Reply:
AFAIK there is a (negligible ?) overhead in the unapply instead of runtime type checking and member access.
The generated unapply should be something like
```object CountAggregation {
  def unapply(c: CountAggregation): Option[String] = Some(c.fieldName)
}```.

On the other hand, for sure, there's no benefit in using the pattern matching, so we can switch to type Checking